### ISSUE TITLE
Plugin does not respect provider.tags configuration

### ISSUE THREAD

**Issue by TylerHendrickson** [DataDog/serverless-plugin-datadog#52]
## Expected Behavior

When `provider.stackTags` are set in Serverless, they should be used to configure tags on individual Lambda function resources within the stack, unless a conflicting tag is set using `provider.tags` or `functions.functionName.tags`, per the README documentation for the `enableTags` setting (emphasis added):

> When set, the plugin will try to automatically tag lambdas with service and env, **but will not override existing tags**. Defaults to true.

Therefore, when I configure Serverless like this:
```yaml
service:
  name: my-service

plugins:
  - serverless-plugin-datadog

custom:
  datadog:
    addLayers: true
    enableTags: true

provider:
  name: aws
  stage: my-stage
  tags:  # Can also be set using stackTags
    env: production
    service: some-service

functions:
  MyFunction:
    handler: functions.handler
    runtime: python3.8
```

I expect that the `MyFunction` Lambda would be tagged with `env: production` and `service: some-service`.

## Actual Behavior

The `MyLambda` plugin is tagged with `env:my-stage` and `service:my-service` because `custom.datadog.enableTags` is set to `true` (the default value). The only way I can ensure that `MyFunction` is tagged with `env: production` and `service: some-service` is if `functions.MyFunction.tags.env` and `functions.MyFunction.tags.service` are set explicitly. For services with more that one function, this quickly becomes a nuisance, especially since the `provider.tags` setting is specifically provided by the Serverless framework for this purpose.

## Steps to Reproduce the Problem

  1. Configure `provider.tags.env` to a value other than that of `provider.stage` and configure `provider.tags.service` to a value other than that of `service.name`.
  1. Do not set `tags.env` or `tags.service` on at least one Lambda function
  1. Ensure that `datadog.enableTags` is not set to `false`
  1. Deploy and inspect the tags of the deployed function(s). You will see that `env:` is set to the value of `provider.stage` and `service:` is set to the value of `service.name`, instead of the values configured at `provider.tags.env` and `provider.tags.service`, respectively.

## Specifications

  - Serverless Framework version: 1.73.1
  - Datadog Serverless Plugin version: 1.0.0
  - Lambda function runtime (Python 3.7, Node 10, etc.): Reproduced using `python3.8` and `node10` runtimes
  
## Stacktrace

  N/A

**Comment by hannah-sun** [comment]
Hi @TylerHendrickson, thanks for opening the issue! We just updated the plugin with a fix (v1.0.1). Let us know if this solves the problem for you!

**Comment by hannah-sun** [comment]
Going to close this for now, feel free to re-open and let us know if the fix doesn't solve the issue!
